snapcraft/internal/elf.py

```diff
@@ -82,21 +125,42 @@ def patch(self, *, elf_file: ElfFile) -> None:
         :raises snapcraft.internal.errors.PatcherError:
             raised when the elf_file cannot be patched.
         """
-        # When setting rpath for libraries is implemented, we will do more
-        # here.
-        if not elf_file.is_executable:
-            return
+        if elf_file.is_executable:
+            self._patch_interpreter(elf_file)
+        if elf_file.dependencies:
+            self._patch_rpath(elf_file)
+
+    def _patch_interpreter(self, elf_file: ElfFile) -> None:
+        self._run_patchelf(['--set-interpreter',  self._dynamic_linker],
+                           elf_file.path)
+
+    def _patch_rpath(self, elf_file: ElfFile) -> None:
+        rpath = self._get_rpath(elf_file)
+        print(elf_file.path, 'rpath: ', rpath)
```

Comment:
This looks like something that should be removed.

Reply:
indeed